martha/martha_v3.js

```diff
@@ -6,156 +6,207 @@ const url = require('url');
-            false,
-            `${config.bondBaseUrl}/api/link/v1/fence/serviceaccount/key`,
-            dosResponseParser);
-    } else if (parsedUrl.host.toLowerCase().startsWith('dg.')) {
+            DG_EXPANSION_BIO_DATA_CATALYST,
+            PROTOCOL_PREFIX_DOS,
+            AUTH_SKIPPED,
+            BOND_PROVIDER_FENCE,
+        );
+    }
+
+    // Full BDC
+    if (host === DG_EXPANSION_BIO_DATA_CATALYST) {
         return new DrsType(
-            dosCompactUrlGenerator(parsedUrl),
-            false,
-            `${config.bondBaseUrl}/api/link/v1/dcf-fence/serviceaccount/key`,
-            dosResponseParser);
-    } else if (parsedUrl.host.endsWith('.dataguids.org')) {
+            DG_EXPANSION_NONE,
+            PROTOCOL_PREFIX_DOS,
+            AUTH_SKIPPED,
+            BOND_PROVIDER_FENCE,
+        );
+    }
+
+    // Compact the AnVIL
+    if (host === 'dg.anv0') {
         return new DrsType(
-            dosCompactUrlGenerator(parsedUrl),
-            false,
-            null,
-            dosResponseParser);
-    } else if ((/jade.*\.datarepo-.*\.broadinstitute\.org/).test(parsedUrl.host)) {
+            DG_EXPANSION_THE_ANVIL,
+            PROTOCOL_PREFIX_DOS,
+            AUTH_SKIPPED,
+            BOND_PROVIDER_ANVIL,
+        );
+    }
+
+    // Full the AnVIL
+    if (host === DG_EXPANSION_THE_ANVIL) {
         return new DrsType(
-            drsFullUrlGenerator(parsedUrl),
-            true,
-            null,
-            drsResponseParser);
-    } else if (parsedUrl.host.endsWith('.humancellatlas.org')) {
+            DG_EXPANSION_NONE,
+            PROTOCOL_PREFIX_DOS,
+            AUTH_SKIPPED,
+            BOND_PROVIDER_ANVIL,
+        );
+    }
+
+    // Full Jade Data Repo
+    if ((/jade.*\.datarepo-.*\.broadinstitute\.org/).test(host)) {
+        return new DrsType(
+            DG_EXPANSION_NONE,
+            PROTOCOL_PREFIX_DRS,
+            AUTH_REQUIRED,
+            BOND_PROVIDER_NONE,
+        );
+    }
+
+    // Full HCA
+    if (host.endsWith('.humancellatlas.org')) {
         return new DrsType(
-            dosFullUrlGenerator(parsedUrl),
-            false,
-            null,
-            dosResponseParser);
-    } else if (parsedUrl.host.endsWith('.datacommons.io')) {
+            DG_EXPANSION_NONE,
+            PROTOCOL_PREFIX_DOS,
+            AUTH_SKIPPED,
+            BOND_PROVIDER_NONE,
+        );
+    }
+
+    // Full CRDC
+    if (host.endsWith('.datacommons.io')) {
         return new DrsType(
-            drsFullUrlGenerator(parsedUrl),
-            true,
-            null,
-            drsResponseParser);
-    } else {
+            DG_EXPANSION_NONE,
+            PROTOCOL_PREFIX_DRS,
+            AUTH_REQUIRED,
+            BOND_PROVIDER_NONE,
+        );
+    }
+
+    // Assume BDC compact but with dcf-fence
+    // If we don't recognize the dg.* assume like martha_v2 that everyone else
+    // speaks DOS, doesn't require auth, and uses dcf-fence
```

Comment:
nit: this comment is hard to read because it contains a sentence fragment all on one line that is capitalized but doesn't have a "." at the end and then a full sentence that is capitalized, split between two lines, and also doesn't have a "." at the end